### PR TITLE
Fix and let `# gazelle:follow` accept a glob

### LINKED ISSUE #1593: language/go/generate_test.go TestGenerateResult spuriously passing
### What version of gazelle are you using?

commit fa6d6eab293df1f27ab6e97e3f6a6821e95306d2

### What version of rules_go are you using?

v0.39.1

### What version of Bazel are you using?

bazel 6.3.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

n/a

### What did you do?

1. Modify `language/go/testdata/bin/BUILD.want` to introduce an erroneous `srcs`
2. Run `bazel test //languge/go:go_test`

### What did you expect to see?

A failure.

### What did you see instead?

Success.

Digging in a bit more, the test harness loops through `regularFiles` provided by `walk.Walk`, but all of the files in the runfiles subdirectory are symlinks, thus skipped by `Walk` (as documented).  The end result is a spurious failure.

### PR DESCRIPTION
**Pull request by shahms**
**What type of PR is this?**
 Bug fix

**What package or component does this PR mostly affect?**
 all (due to the implementation of the `gazelle:follow` directive)

**What does this PR do? Why is it needed?**
This serves as a proposed resolution for #1593 but also actually implements the `follow` directive to do so.

**Which issues(s) does this PR fix?**

Fixes #1593 

**Other notes for review**

### CODE REVIEW

File: walk/config.go [hunk 1]
```diff
@@ -108,6 +98,10 @@ func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
 				}
 				wcCopy.excludes = append(wcCopy.excludes, path.Join(rel, d.Value))
 			case "follow":
+				if err := checkPathMatchPattern(path.Join(rel, d.Value)); err != nil {
```
Comment (fmeum) [inline walk/config.go:101]: Am I right to say that this has very minor potential for breaking existing uses, namely only if a file path that was previously interpreted literally contains characters with special meaning for `doublestar.Match`? That would seem acceptable, but we should probably mention the set of these characters and how to escape them in the docs for `# gazelle:follow`.

**Review by fmeum — APPROVED** [review]
@linzhp Are you using `# gazelle:follow` at Uber?

File: walk/config.go (hunk 1 above)
Comment (shahms) [reply walk/config.go:101]: I should note that prior to this change `follow` was set on `walkConfig` but never actually read or used, so once potentially source of breakage is extant uses of `follow` that now become active.

**Comment by linzhp** [comment]
> @linzhp Are you using `# gazelle:follow` at Uber?

No we don't

**Review by fmeum — APPROVED** [review]
Thanks!

**Pull request merged**